File: frontend/src/app/common/utils/location-status.ts
/** Statut de localisation d'un scan (LocationStatus du back). */
export type LocationStatus =
  | 'VERIFIED'
  | 'TOO_FAR'
  | 'NO_GPS'
  | 'IMPRECISE'
  | 'TAG_NOT_CALIBRATED';

/** Libellé français par statut, pour le back-office. */
export const LOCATION_LABEL: Record<LocationStatus, string> = {
  VERIFIED: 'Position vérifiée',
  TOO_FAR: 'Trop loin du tag',
  NO_GPS: 'Sans GPS',
  IMPRECISE: 'GPS imprécis',
  TAG_NOT_CALIBRATED: 'Tag non calibré'
};

File: backend/src/main/java/com/nfctag/features/location/LocationStatus.java
package com.nfctag.features.location;

public enum LocationStatus {
    /** GPS fiable et à portée du tag : passage confirmé. */
    VERIFIED,
    /** GPS fiable mais loin du tag : anomalie à contrôler. */
    TOO_FAR,
    /** Localisation refusée ou indisponible sur le téléphone. */
    NO_GPS,
    /** GPS trop imprécis pour conclure (fréquent en intérieur). */
    IMPRECISE,
    /** Le tag n'a pas encore de position enregistrée. */
    TAG_NOT_CALIBRATED
}
